pkg/install/kamelets.go

```diff
@@ -19,30 +19,40 @@ package install
 
 import (
 	"context"
+	"errors"
 	"fmt"
-	"io/ioutil"
+	"io/fs"
+	"net/http"
 	"os"
 	"path"
+	"path/filepath"
 	"strings"
 
-	"github.com/apache/camel-k/pkg/util"
+	"golang.org/x/sync/errgroup"
 
-	"github.com/pkg/errors"
 	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
 	"k8s.io/apimachinery/pkg/types"
 
+	ctrl "sigs.k8s.io/controller-runtime/pkg/client"
+
 	"github.com/apache/camel-k/pkg/apis/camel/v1alpha1"
 	"github.com/apache/camel-k/pkg/client"
+	"github.com/apache/camel-k/pkg/util"
 	"github.com/apache/camel-k/pkg/util/defaults"
 	"github.com/apache/camel-k/pkg/util/kubernetes"
+	"github.com/apache/camel-k/pkg/util/patch"
 )
 
 const (
 	kameletDirEnv     = "KAMELET_CATALOG_DIR"
 	defaultKameletDir = "/kamelets/"
 )
 
-// KameletCatalog installs the bundled KameletCatalog into one namespace.
+var hasServerSideApply = true
```

Comment:
Should this be `atomic.Value`?

Reply:
It should be set the first time by the main goroutine only. Even if this races, Server-Side Apply will be tried twice, but my understanding is that it's idempotent. Also the client-side logic will ultimately be removed as SSA becomes ubiquitous, so I guess this explains why I did not bother using an atomic boolean. WDYT?

Reply:
Yeah in the grand scheme of things it could only be a performance issue, i.e. trying SSA multiple times, not a logic issue. My preference would be to use atomic value here, but I'll leave it up to you...